### ISSUE TITLE
Not working focus on iOS

### ISSUE THREAD

**Issue by Jekins** [shentao/vue-multiselect#717]
Not working focus on iOS if mode searchable is true. After click on multiselect, input not set focused.

**Comment by chocochaii** [comment]
As I researched, you have to click 2 times to get focus state on iOS. The first click is on <span> so the options will open but the keyboard doesn't. If we want to open the keyboard, we have to click it again. This second time we actually click on <input> so the keyboard will open.

**Comment by Jekins** [comment]
But if I need to open the keyboard with the first click, what should I do? Because that's how all the selections work and this is how this plugin works on android

**Comment by darinkolev** [comment]
here is a workaround:

`multiselect(@open="onSelect")`

and on select show and focus programmatically the input:

` 
onSelect () {
      const input = $(this.$vnode.elm).find('input')
      if (input.css('display') === 'none') {
        input.show()
        input.trigger('focus')
      }
    },
`

you have to adjust the selector of the input

**Comment by Jekins** [comment]
Thx, it is work:
Set ```ref="multiselect"``` and ```@open="open"``` on select, and writing this method:
```
open () {
      const multiselect = this.$refs.multiselect
      const input = multiselect.$refs.search

      if (input.style.display === 'none') {
        input.style.display = 'block'
        input.focus()
      }
}
```

**Comment by Syther101** [comment]
This still seems to be an issue in Safari on iOS 12. I've captured the below video detailing the issue.

![vue-safari-bug](https://user-images.githubusercontent.com/1341001/52065152-dbc08100-256d-11e9-8c93-e8a70b56e8a9.gif)

Looking at links around the web this seems to be an issue with simulating focus on the input element. Safari seems to not respond to simulated clicks like other browsers.

[iOS does not show keyboard on .focus()](https://github.com/jquery/jquery-mobile/issues/3016)

[Mobile Safari: Javascript focus() method on inputfield only works with click?
](https://stackoverflow.com/questions/12204571/mobile-safari-javascript-focus-method-on-inputfield-only-works-with-click/16601288)

I've managed to fix this in a project by removing the `div.multiselect__placeholder` element and simply moving the `input.multiselect__input` element to the front of the component. Doing this means Safari will correctly receive focus as the touch event happens on the input element itself.

@shentao I'n not sure if the `div.multiselect__placeholder` element has other purposes. If it could be removed and replaced with the above fix I could submit a pull request 🙂

**Comment by shentao** [comment]
Thanks @syther101 for the investigation. What about not removing the placeholder, but instead moving the z-index of the input so that it is on-top of the placeholder?

**Comment by shentao** [comment]
Thanks @syther101 for the investigation. What about not removing the placeholder, but instead moving the z-index of the input so that it is on-top of the placeholder?

**Comment by syther101** [comment]
Agreed @shentao. I've given it a try and it seems to work nicely. I don't use the default multiselect css in this project so will try and work creating a PR in the coming week :)

**Comment by shentao** [comment]
Thanks a lot!
